src/components/views/rooms/MessageComposerInput.js

```diff
@@ -190,11 +190,11 @@ export default class MessageComposerInput extends React.Component {
                 const Pill = sdk.getComponent('elements.Pill');
                 const {url} = Entity.get(props.entityKey).getData();
                 if (Pill.isPillUrl(url)) {
-                    return <Pill url={url} room={this.props.room}/>;
+                    return <Pill url={url} room={this.props.room} offsetKey={props.offsetKey}/>;
```

Comment:
Eep, took me a while to realise you have a functional component nested inside a component and therefore have two props in scope, on at this.props and one at props...